analytics/google_analytics.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package analytics
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+
+	"github.com/google/go-querystring/query"
+	"github.com/openfaas/faas-cli/version"
+)
+
+const gaHost = "www.google-analytics.com"
+const trackingID = "UA-107707760-3"
+const applicationName = "faas-cli3"
+
+// disableEnvvar will prevent submission of analytics events if set
+const disableEnvvar = "OPEN_FAAS_TELEMETRY"
+
+// Event posts an analytics event to GA
+func Event(action string, language string, ch chan int) {
+	if Disabled() {
+		return
+	}
+	u, err := NewSession(language)
+	if err != nil {
+		return
+	}
+	u.EventAction = action
+	go u.PostEvent(ch)
+}
+
+// NewSession provides a setup UserSession struct with sane defaults
+func NewSession(language string) (*UserSession, error) {
+	if len(language) == 0 {
+		language = "unset"
+	}
+	userSession := &UserSession{
+		HTTPClient:         http.DefaultClient,
+		ProtocolVersion:    1,
+		Type:               "event",
+		TrackingID:         trackingID,
+		ApplicationName:    applicationName,
+		ApplicationVersion: version.BuildVersion(),
+		AnonymizeIP:        1,
+		Language:           language,
+		OS:                 runtime.GOOS,
+		ARCH:               runtime.GOARCH,
```

Comment:
Just a thought this may mis-report I think it's set at build-time. One thing it won't do is to show armv6/7/8 differences. For that an OS library like pstate etc might help - although we'd need one that doesn't use CGO.